custom/icds/location_rationalization/parser.py

```diff
@@ -0,0 +1,74 @@
+from collections import defaultdict
+
+from custom.icds.location_rationalization.const import (
+    EXTRACT_TRANSITION,
+    MERGE_TRANSITION,
+    MOVE_TRANSITION,
+    SPLIT_TRANSITION,
+    TRANSITION_COLUMN_NAME,
+    VALID_TRANSITIONS,
+)
+
+
+class Parser(object):
+    def __init__(self, worksheet, location_types):
+        self.worksheet = worksheet
+        self.requested_transitions = {}
+        self.errors = []
+        self.location_types = location_types
+        self.transitions = {location_type: {
```

Comment:
I have added some documentation to clarify the structure to the `Parser` class and along with the naming changes i did, things should be easy to understand now.
I am hoping this would be a small list of operations only considering the scale of changes to consider, but anyway, I can look to improve this if you think this should still be improved.
I understand the dict structure is a bit weird.

> Another option would be to create small data-only classes
This would be a good change but i would like to not block on it.

Reply:
Hmm. I still think it'd be better for the structure to be less weird. I'm okay with moving forward with merging this PR and seeing how the rest of this feature interacts with this structure, provided you're okay with me possibly complaining about it again later on.

Reply:
Yes, i will be. :)
Thank you.